File: src/pocketcloud/network/packet/ResponsePacket.php
<?php

namespace pocketcloud\network\packet;

use pocketcloud\network\client\ServerClient;
use pocketcloud\network\packet\utils\PacketData;

abstract class ResponsePacket extends CloudPacket {

    private string $requestId = "";

    public function encode(PacketData $packetData): void {
        parent::encode($packetData);
        $packetData->write($this->requestId);
    }

    public function decode(PacketData $packetData): void {
        parent::decode($packetData);
        $this->requestId = $packetData->readString();
    }

    public function getRequestId(): string {
        return $this->requestId;
    }

    public function setRequestId(string $requestId): self {
        $this->requestId = $requestId;
        return $this;
    }

    final public function handle(ServerClient $client) {}
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/ResponsePacket.java
package de.pocketcloud.network.packet;

import de.pocketcloud.api.network.packet.data.IPacketData;
import lombok.Getter;
import lombok.Setter;

public abstract class ResponsePacket extends CloudPacket {

    @Getter
    @Setter
    private String requestId = "";

    @Override
    public final void encode(IPacketData packetData) {
        super.encode(packetData);
        packetData.write(requestId);
    }

    @Override
    public final void decode(IPacketData packetData) {
        super.decode(packetData);
        requestId = packetData.readString();
    }
}